src/main/java/sg/ncl/MainController.java

```diff
@@ -3494,6 +3505,531 @@ public String adminEnergy(Model model,
+            }
+            List<ProjectDetails> projectsList = getProjects();
+
+            for (ProjectDetails project : projectsList) {
+                // compute active and inactive projects
+                differentiateProjects(newProjects, activeProjects, inactiveProjects, m_s, m_e, project);
+
+                // monthly utilisation
+                computeMonthlyUtilisation(utilizationMap, formatter, m_s, m_e, project);
+
+                // usage statistics by category
+                totalCategoryUsage += getCategoryUsage(statsCategoryMap, m_s, m_e, project);
+
+                // usage statistics by academic institutes
+                totalAcademicUsage += getAcademicUsage(statsAcademicMap, m_s, m_e, project);
+            }
+        }
+
+        attributes.addFlashAttribute(KEY_QUERY, query);
+        attributes.addFlashAttribute("newProjects", newProjects);
+        attributes.addFlashAttribute("activeProjects", activeProjects);
+        attributes.addFlashAttribute("inactiveProjects", inactiveProjects);
+        attributes.addFlashAttribute("months", months);
+        attributes.addFlashAttribute("utilization", utilizationMap);
+        attributes.addFlashAttribute("statsCategory", statsCategoryMap);
+        attributes.addFlashAttribute("totalCategoryUsage", totalCategoryUsage);
+        attributes.addFlashAttribute("statsAcademic", statsAcademicMap);
+        attributes.addFlashAttribute("totalAcademicUsage", totalAcademicUsage);
+
+        return "redirect:/admin/statistics";
+    }
+
+    private void differentiateProjects(List<ProjectDetails> newProjects,
+                                       List<ProjectDetails> activeProjects,
+                                       List<ProjectDetails> inactiveProjects,
+                                       YearMonth m_s, YearMonth m_e,
+                                       ProjectDetails project) {
+        YearMonth created = YearMonth.from(project.getZonedDateCreated());
+        YearMonth m_e_m2 = m_e.minusMonths(2);
+        YearMonth m_active = m_e_m2.isBefore(m_s) ? m_e_m2 : m_s;
+
+        // projects created within the period
+        if (!(created.isBefore(m_s) || created.isAfter(m_e))) {
+            newProjects.add(project);
+        }
+
+        // active projects = projects with resources within the period + projects created
+        boolean hasUsage = project.getProjectUsages().stream().anyMatch(p -> p.hasUsageWithinPeriod(m_active, m_e));
+        if (hasUsage || !(created.isBefore(m_e_m2) || created.isAfter(m_e))) {
+            activeProjects.add(project);
+        }
+
+        // inactive projects
+        if (!hasUsage && created.isBefore(m_e_m2)) {
+            inactiveProjects.add(project);
+        }
+    }
+
+    private void computeMonthlyUtilisation(Map<String, MonthlyUtilization> utilizationMap, DateTimeFormatter formatter, YearMonth m_s, YearMonth m_e, ProjectDetails project) {
+        YearMonth counter = m_s;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)